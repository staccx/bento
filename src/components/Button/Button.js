import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import {
  color,
  borderRadius,
  spacing,
  targetSize,
  font,
  fontFamily,
  fontWeight,
  applyVariants,
  commonPropTypes
} from "../../theming"
import themeProps from "./Button.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"

const ButtonComp = styled.button`
  background-color: ${color.primary};
  color: ${color.white};
  border-width: 0;
  border-radius: ${borderRadius};
  padding: ${spacing.tiny} ${spacing.medium};
  cursor: pointer;
  min-height: ${targetSize.normal};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  line-height: 32px;
  font-weight: ${fontWeight.bold};
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  transition: background 0.2s ease;

  &:hover,
  &:focus {
    outline: none;
    background-color: ${color.primary};
    color: ${color.white};
  }

  ${applyVariants(themeProps.button)};
`
/**
 * Button component
 */
const Button = props => <ButtonComp {...props} />

const defaultProps = {
  children: "button",
  onClick: null
}

const propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Click callback. Lets you do stuff onClick
   * @export
   */
  onClick: PropTypes.func
}

Button.defaultProps = defaultProps
Button.propTypes = propTypes
Button.themeProps = themeProps
Button.createVariants = componentCreateFactory(Button)

export default Button
