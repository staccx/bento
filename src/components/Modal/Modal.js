import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import IconClose from "../Icons/Close/Close"
import { FadeIn } from "../../animations"
import {
  applyVariants,
  borderRadius,
  color,
  commonPropTypes,
  hideVisually,
  spacing,
  wrapper
} from "../../theming"
import ThemeComponent from "../Theme/ThemeComponent/ThemeComponent"
import themeProps from "./Modal.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"
import { useFocusTrap } from "../../hooks/useFocusTrap/useFocusTrap"

const Modal = ({ children, className, isOpen, onClose, variant, ...props }) => {
  const [open, setOpen] = React.useState(false)
  const containerRef = React.useRef(null)
  const { ready, activate } = useFocusTrap(
    containerRef,
    {
      activateOnReady: false
    },
    {
      fallbackFocus: "#modal"
    }
  )

  const handleClose = event => {
    if (onClose) {
      onClose(event)
    }

    setOpen(!open)
  }

  const fixOverflow = () => {
    document.documentElement.style.overflow = open ? "hidden" : ""
  }

  React.useEffect(() => {
    const escFunction = event => {
      if (event.keyCode === 27) {
        if (onClose) {
          onClose(event)
        }
        setOpen(false)
      }
    }
    document.addEventListener("keydown", escFunction)
    return () => {
      document.removeEventListener("keydown", escFunction)
      document.documentElement.style.overflow = ""
    }
  }, [])

  React.useEffect(() => {
    fixOverflow()
    if (open && ready) {
      activate()
    }
  }, [open, ready])
  React.useEffect(() => {
    if (isOpen !== open) {
      setOpen(isOpen)
    }
  }, [isOpen])

  return (
    <div ref={containerRef}>
      {open && (
        <>
          <ModalItem
            className={className}
            {...props}
            role="dialog"
            tabIndex="0"
            open="open"
            aria-labelledby="modal"
            variant={variant}
          >
            <ModalContent
              role="document"
              tabIndex="0"
              id="modal"
              variant={variant}
            >
              <Close
                type="button"
                id="modal-close"
                aria-label="Close (Press escape to close)"
                onClick={handleClose}
                variant={variant}
              >
                <span>Close</span>
                <Icon />
              </Close>
              {children}
            </ModalContent>
          </ModalItem>
          <ModalBackdrop onClick={handleClose} variant={variant} />
        </>
      )}
    </div>
  )
}

const ModalItem = styled.dialog`
  border-width: 0;
  height: 100%;
  width: 100%;
  overflow: auto;
  position: fixed;
  pointer-events: none;
  top: 0;
  left: 0;
  z-index: 9999;
  display: grid;
  grid-template-columns: 24px 1fr 1fr 24px;
  grid-gap: 12px;
  grid-template-areas:
    ". . . ."
    ". a a ."
    ". a a ."
    ". . . .";
  background: transparent;
  padding: ${spacing.large} ${spacing.small} ${spacing.small} ${spacing.small};
  &:focus {
    outline: none;
  }
  ${applyVariants(themeProps.dialog)};
`

const ModalContent = styled.div`
  background: white;
  border-radius: ${borderRadius};
  grid-area: a;
  align-self: center;
  justify-self: center;
  padding: ${spacing.large} ${spacing.medium} ${spacing.medium}
    ${spacing.medium};
  max-width: ${wrapper.medium};
  min-width: 296px;
  width: 100%;
  overflow-y: auto;
  position: relative;
  pointer-events: all;
  &:focus {
    outline: none;
  }
  ${applyVariants(themeProps.content)};
`

const Close = styled.button`
  position: absolute;
  top: 0;
  right: 0;
  z-index: 9999;
  background: transparent;
  border-width: 0;
  color: ${color.primary};
  padding: ${spacing.small};
  cursor: pointer;
  &:focus,
  &:hover {
    outline: none;
    color: ${color.secondary};
  }
  span {
    ${hideVisually};
  }
  ${applyVariants(themeProps.close)};
`

const ModalBackdrop = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.8);
  opacity: 0;
  animation: 0.2s ${FadeIn} ease-out forwards 1;
  ${applyVariants(themeProps.backdrop)};
`

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={themeProps.closeIcon}
    fallback={IconClose}
    {...props}
  />
)

const Icon = styled(IconComponent)``

Modal.defaultProps = {
  isOpen: false,
  className: ""
}

Modal.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  /**
   * When isOpen is true, the modal is displayed
   */
  isOpen: PropTypes.bool,
  onClose: PropTypes.func
}
Modal.themeProps = themeProps
Modal.createVariants = componentCreateFactory(Modal)

export default Modal
