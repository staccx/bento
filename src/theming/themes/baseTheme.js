import React from "react"
import Theme from "../Theme"
import reset from "../styles/cssResets"
import grids from "../../components/Layout/Layout/Layout.Grids"
import codeTheme from "../codeTheme"

const Logo = () => <div>ThemeComponent for BaseTheme</div>
Logo.displayName = "Logo for base theme"

export const theme = {
  name: "Base Theme",
  storybookName: "Base Theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190", // for toggle eks
    link: "#279AF1"
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px",
    lineHeight: "1.5"
  },
  fontWeight: {
    bold: "bold",
    normal: "normal",
    light: "lighter"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px",
  reset,
  grid: grids,
  logo: Logo
}

export const StyleguideTheme = new Theme(theme, codeTheme)

export default theme
