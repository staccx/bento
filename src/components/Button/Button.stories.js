import React from "react"
import Button from "./Button"

export default {
  title: "Components/Button",
  component: Button
}

export const text = () => <Button> Button </Button>
