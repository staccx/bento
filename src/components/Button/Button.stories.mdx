import {useState} from "react"
import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { action }Â from "@storybook/addon-actions"
import Button from "./Button"

<Meta title="components/Button" component={Button} parameters={{component: Button.themeProps.button.name}} />

# Button
Bento Button Component

```javascript
import {Button} from "@staccx/bento"
```


## Standard Button
This is the standard button

<Preview>
  <Story name="Standard">
   <Button>Knapp</Button>
  </Story>
</Preview>


## Long String
<Preview>
  <Story name="Long String">
   <Button>This is a button with long text</Button>
  </Story>
</Preview>

## With useState
<Preview>
  <Story name="with counter">
    {() => {
      const [counter, setCounter] = useState(0);
      const label = `Testing: ${counter}`;
      return <Button onClick={() => setCounter(counter + 1)}>{label}</Button>;
    }}
  </Story>
</Preview>

### Source code

<details><summary>Button</summary>
<p>

```bash
import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import {
  color,
  borderRadius,
  spacing,
  targetSize,
  font,
  fontFamily,
  fontWeight,
  applyVariants,
  commonPropTypes
} from "../../theming"
import themeProps from "./Button.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"

const ButtonComp = styled.button`
  background-color: ${color.primary};
  color: ${color.white};
  border-width: 0;
  border-radius: ${borderRadius};
  padding: ${spacing.tiny} ${spacing.medium};
  cursor: pointer;
  min-height: ${targetSize.normal};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  line-height: 32px;
  font-weight: ${fontWeight.bold};
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  transition: background 0.2s ease;

  &:hover,
  &:focus {
    outline: none;
    background-color: ${color.primary};
    color: ${color.white};
  }

  ${applyVariants(themeProps.button)};
`
/**
 * Button component. Use to click stuff
 */
const Button = props => <ButtonComp {...props} />

const defaultProps = {
  children: "button",
  onClick: null
}

const propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Click callback. Lets you do stuff onClick
   * @export
   */
  onClick: PropTypes.func
}

Button.defaultProps = defaultProps
Button.propTypes = propTypes
Button.themeProps = themeProps
Button.createVariants = componentCreateFactory(Button)

export default Button

```

</p>
</details>


## Props for button
<Props of={Button} />
