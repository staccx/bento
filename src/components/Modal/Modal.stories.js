import React, { useState } from "react"
import Modal from "./Modal"
import Button from "../Button/Button"

export default {
  title: "Components/Modal",
  component: Modal
}

export const Standard = args => {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <div>
      <Button onClick={() => setIsOpen(true)}>Open Modal</Button>
      <Modal isOpen={isOpen} onClose={() => setIsOpen(false)}>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque
          iaculis erat ipsum, ut euismod orci luctus sit amet. Donec et
          tincidunt dolor, at vehicula mauris. Nullam eleifend tristique arcu ac
          scelerisque. Pellentesque sapien ex, faucibus sit amet nulla eget,
          sollicitudin fringilla arcu. Suspendisse imperdiet urna eu nulla
          tincidunt pretium. Nunc a blandit nulla, ac feugiat nulla. Nulla nisl
          nunc, ornare quis odio gravida, sodales mattis enim. Aenean turpis
          dui, fringilla in arcu a, consectetur placerat sapien. Aenean nec
          augue quis ipsum feugiat maximus. Interdum et malesuada fames ac ante
          ipsum primis in faucibus. Suspendisse vitae nisl vehicula ante gravida
          scelerisque faucibus ac enim. Suspendisse imperdiet tellus id
          tristique scelerisque.
        </p>
        <p>
          Ut rutrum ex ante, ut posuere arcu pellentesque vel. Mauris sodales
          urna enim, a sodales lorem fringilla sit amet. Vivamus ultricies non
          mauris ac maximus. Donec sit amet eleifend nibh, eu mollis lorem.
          Praesent tincidunt, mauris pharetra facilisis molestie, urna nisi
          cursus enim, vitae congue ligula eros tempor felis. Sed sit amet felis
          nulla. Quisque dictum aliquet quam vitae venenatis. Donec eu nibh quis
          magna auctor cursus. Duis malesuada arcu mi, facilisis accumsan risus
          iaculis eu. Sed a lacinia mauris.
        </p>
        <button>This is focused firstly</button>
        <button>This is focused secondly</button>
      </Modal>
    </div>
  )
}
