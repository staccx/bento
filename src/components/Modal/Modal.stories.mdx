import {useState} from "react"
import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Modal from "./Modal"
import { action }Â from "@storybook/addon-actions"
import Button from "../Button/Button"

<Meta title="components/Modal" component={Modal}/>

# Modal
Modal is a dialog box/popup window that is displayed on top of the current page

```javascript
import { Modal } from "@staccx/bento"
```


## Standard Modal
Modal is open, and closes if you click outside the modal. Klick the button to reopen the modal.



  <Story name="standard">
   {() => {
    const [isOpen, setIsOpen] = useState(false);
    return <div>
    <Button onClick={() => setIsOpen(true)}>Open Modal</Button>
    <Modal isOpen={isOpen} onClose={() => setIsOpen(false)}>
                   <p>
                   {" "}
                   Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque
                   iaculis erat ipsum, ut euismod orci luctus sit amet. Donec et
                   tincidunt dolor, at vehicula mauris. Nullam eleifend tristique arcu
                   ac scelerisque. Pellentesque sapien ex, faucibus sit amet nulla
                   eget, sollicitudin fringilla arcu. Suspendisse imperdiet urna eu
                   nulla tincidunt pretium. Nunc a blandit nulla, ac feugiat nulla.
                   Nulla nisl nunc, ornare quis odio gravida, sodales mattis enim.
                   Aenean turpis dui, fringilla in arcu a, consectetur placerat sapien.
                   Aenean nec augue quis ipsum feugiat maximus. Interdum et malesuada
                   fames ac ante ipsum primis in faucibus. Suspendisse vitae nisl
                   vehicula ante gravida scelerisque faucibus ac enim. Suspendisse
                   imperdiet tellus id tristique scelerisque.
                 </p>
                 <p>
                   Ut rutrum ex ante, ut posuere arcu pellentesque vel. Mauris sodales
                   urna enim, a sodales lorem fringilla sit amet. Vivamus ultricies non
                   mauris ac maximus. Donec sit amet eleifend nibh, eu mollis lorem.
                   Praesent tincidunt, mauris pharetra facilisis molestie, urna nisi
                   cursus enim, vitae congue ligula eros tempor felis. Sed sit amet
                   felis nulla. Quisque dictum aliquet quam vitae venenatis. Donec eu
                   nibh quis magna auctor cursus. Duis malesuada arcu mi, facilisis
                   accumsan risus iaculis eu. Sed a lacinia mauris.
                 </p>
                 <button>This is focused first</button>
                 <button>This is focused secondly</button>
       </Modal>
    </div>
   }}
  </Story>

### Source code
<details><summary>Modal</summary>
<p>

```bash
import React, { Component } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import IconClose from "../Icons/Close/Close"
import { FadeIn } from "../../animations"
import {
  spacing,
  color,
  wrapper,
  applyVariants,
  borderRadius,
  commonPropTypes,
  hideVisually
} from "../../theming"
import ThemeComponent from "../Theme/ThemeComponent"
import themeProps from "./Modal.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"

class Modal extends Component {
  constructor(props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.escFunction = this.escFunction.bind(this)
    this.state = {
      isOpen: this.props.isOpen
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.isOpen !== this.state.isOpen) {
      this.setState({ isOpen: nextProps.isOpen })
    }
  }

  escFunction(event) {
    if (event.keyCode === 27) {
      if (this.props.onClose) {
        this.props.onClose(event)
      }
      this.setState({
        isOpen: false
      })
    }
  }

  handleChange(event) {
    if (this.props.onClose) {
      this.props.onClose(event)
    }
    // TODO: Make controllable/uncontrollable toggle.
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  fixOverflow(event) {
    if (this.state.isOpen) {
      document.documentElement.style.overflow = "hidden"
    } else {
      document.documentElement.style.overflow = ""
    }
  }

  componentDidMount() {
    document.addEventListener("keydown", this.escFunction, false)
  }

  componentWillUnmount() {
    document.removeEventListener("keydown", this.escFunction, false)
  }

  render() {
    const { children, className, variant, ...otherProps } = this.props
    const { isOpen } = this.state
    this.fixOverflow()
    if (isOpen) {
      return (
        <>
          <ModalItem
            className={className}
            {...otherProps}
            role="dialog"
            tabIndex="0"
            open="open"
            aria-labelledby="modal"
            variant={variant}
          >
            <ModalContent
              role="document"
              tabIndex="0"
              id="modal"
              variant={variant}
            >
              <Close
                type="button"
                id="modal-close"
                aria-label="Close (Press escape to close)"
                onClick={this.handleChange}
              >
                <span>Close</span>
                <Icon />
              </Close>
              {children}
            </ModalContent>
          </ModalItem>
          <ModalBackdrop onClick={this.handleChange} variant={variant} />
        </>
      )
    }
    return null
  }
}

const ModalItem = styled.dialog`
  border-width: 0;
  height: 100%;
  width: 100%;
  overflow: auto;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 9999;
  display: grid;
  grid-template-columns: 24px 1fr 1fr 24px;
  grid-gap: 12px;
  grid-template-areas:
    ". . . ."
    ". a a ."
    ". a a ."
    ". . . .";
  background: transparent;
  padding: ${spacing.large} ${spacing.small} ${spacing.small} ${spacing.small};
  &:focus {
    outline: none;
  }
  ${applyVariants(themeProps.dialog)};
`

const ModalContent = styled.div`
  background: white;
  border-radius: ${borderRadius};
  grid-area: a;
  align-self: center;
  justify-self: center;
  padding: ${spacing.large} ${spacing.medium} ${spacing.medium}
    ${spacing.medium};
  max-width: ${wrapper.medium};
  min-width: 296px;
  width: 100%;
  overflow-y: auto;
  position: relative;
  &:focus {
    outline: none;
  }
  ${applyVariants(themeProps.content)};
`

const Close = styled.button`
  position: absolute;
  top: 0;
  right: 0;
  z-index: 9999;
  background: transparent;
  border-width: 0;
  fill: ${color.primary};
  padding: ${spacing.small};
  cursor: pointer;
  &:focus,
  &:hover {
    outline: none;
    fill: ${color.primary};
  }
  span {
    ${hideVisually};
  }
  ${applyVariants(themeProps.close)};
`

const ModalBackdrop = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.8);
  opacity: 0;
  animation: 0.2s ${FadeIn} ease-out forwards 1;
  ${applyVariants(themeProps.backdrop)};
`

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={themeProps.closeIcon}
    fallback={IconClose}
    {...props}
  />
)

const Icon = styled(IconComponent)``

Modal.defaultProps = {
  isOpen: false,
  className: ""
}

Modal.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  /**
   * When isOpen is true, the modal is displayed
   */
  isOpen: PropTypes.bool,
  onClose: PropTypes.func
}
Modal.themeProps = themeProps
Modal.createVariants = componentCreateFactory(Modal)

export default Modal

```

</p>
</details>

## Props for Modal
<Props of={Modal} />



