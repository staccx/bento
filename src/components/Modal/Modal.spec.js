import React from "react"
import renderer from "react-test-renderer"
import { fireEvent, render, screen, waitFor } from "@testing-library/react"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Modal from "./Modal"
import Button from "../Button/Button"

describe("Modal", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Modal>
              <p> Lorem ipsum dolor sit amet</p>
              <p>Ut rutrum ex ante, ut posuere arcu pellentesque vel.</p>
            </Modal>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Modal className="className">
              <p> Lorem ipsum dolor sit amet</p>
              <p>Ut rutrum ex ante, ut posuere arcu pellentesque vel.</p>
            </Modal>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("isOpen", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Modal className="className" isOpen>
              <p> Lorem ipsum dolor sit amet</p>
              <p>Ut rutrum ex ante, ut posuere arcu pellentesque vel.</p>
            </Modal>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onClose", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Modal
              className="className"
              onClose={e => console.log(e.target.value)}
            >
              <p> Lorem ipsum dolor sit amet</p>
              <p>Ut rutrum ex ante, ut posuere arcu pellentesque vel.</p>
            </Modal>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
  describe("Functionality", () => {
    it("Should open", async () => {
      const testRenderer = renderer.create(
        <Modal isOpen>
          <div className="testValue">Tester</div>
        </Modal>
      )

      await waitFor(() => {
        expect(testRenderer.root.props.isOpen).toBe(true)
        expect(
          testRenderer.root.findByProps({ className: "testValue" }).children
        ).toStrictEqual(["Tester"])

        expect(testRenderer.root.children.length).toBe(1)
      })
    })

    it("Should render empty child if not open", () => {
      const testRenderer = renderer.create(
        <Modal isOpen={false}>
          <div className="testValue">Tester</div>
        </Modal>
      )

      expect(testRenderer.root.props.isOpen).toBe(false)
      expect(testRenderer.root.children.length).toBe(1)
    })
  })
  describe("Rendering", () => {
    it("should close on ESC", async () => {
      render(
        <ThemeProvider theme={baseTheme}>
          <Modal isOpen onClose={() => console.log("closed")}>
            <div className="testValue">Tester</div>
            <Button>Focusable</Button>
          </Modal>
        </ThemeProvider>
      )
      await waitFor(
        () => {
          fireEvent.keyPress(screen.getByText("Tester"), {
            key: "Esc",
            code: "Esc"
          })
          expect(screen.queryBy * "Tester").toBeNaN()
        },
        {
          timeout: 1000
        }
      )
    })
  })
})
