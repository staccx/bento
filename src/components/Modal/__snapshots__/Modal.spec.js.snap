// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshots No props 1`] = `<div />`;

exports[`Modal Snapshots className 1`] = `<div />`;

exports[`Modal Snapshots isOpen 1`] = `<div />`;

exports[`Modal Snapshots onClose 1`] = `<div />`;
